client/request.go

```diff
@@ -184,10 +184,10 @@ func (cli *Client) doRequest(req *http.Request) (serverResponse, error) {
 		// `open //./pipe/docker_engine: Le fichier spécifié est introuvable.`
 		if strings.Contains(err.Error(), `open //./pipe/docker_engine`) {
 			// Checks if client is running with elevated privileges
-			if f, elevatedErr := os.Open("\\\\.\\PHYSICALDRIVE0"); elevatedErr == nil {
+			if f, elevatedErr := os.Open(`\\.\PHYSICALDRIVE0`); elevatedErr == nil {
+				_ = f.Close()
 				err = errors.Wrap(err, "in the default daemon configuration on Windows, the docker client must be run with elevated privileges to connect")
 			} else {
-				f.Close()
 				err = errors.Wrap(err, "this error may indicate that the docker daemon is not running")
```

Comment:
I'm actually wondering if the condition as a whole is flipped, but I don't have a Windows machine to try;


From reading the code, I _THINK_ an error would be returned if the use is _NOT_ privileged; in that case, opening `\\.\PHYSICALDRIVE0` would fail, and the "try running with elevated privileges" would be the error to show.

If that does _not_ fail, and the most likely reason for the original error was that the daemon was not running.


All that said, I wonder if this whole code should be handled elsewhere; after all, we should _know_ how the client is trying to connect, so I assume we can check there if the user has privileges to open the named pipe?

Reply:
Looking at the 'How to verify' section in https://github.com/moby/moby/pull/39588#issue-470987143, I'm pretty sure you're right. The code doesn't match the intended behavior.

Once you replace `elevatedErr == nil` with `elevatedErr != nil`, `f.Close()` is located in the right branch. Maybe worth fixing that instead?

Reply:
Yup! Thanks for confirming my suspicion; I updated the PR with that change; PTAL